lib/ssl/doc/src/using_ssl.xml

```diff
@@ -119,74 +126,77 @@ ok</code>
       
-      <p><em>Step 6:</em> Do the TLS handshake:</p>
-      <code type="erl">5 server> {ok, TLSSocket} = ssl:handshake(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}]).
-{ok,{sslsocket,[...]}}</code>
+  <p>To upgrade to a TLS connection:</p>
+
+  <p><em>Step 1:</em> Start the server side:</p>
+  <code type="erl">1 server> ssl:start().
+  ok</code>
+
+  <p><em>Step 2:</em> Create a normal TCP listen socket and ensure
+  <c>active</c> is set to <c>false</c> and not set to any active mode
+  otherwise TLS handshake messages can be delivered to the
+  wrong process.
+  </p>
+  <code type="erl">2 server> {ok, ListenSocket} = gen_tcp:listen(9999, [{reuseaddr, true},
+  {active, false}]).
+  {ok, #Port&lt;0.475&gt;}</code>
+
+  <p><em>Step 3:</em> Accept client connection:</p>
+  <code type="erl">3 server> {ok, Socket} = gen_tcp:accept(ListenSocket).
+  {ok, #Port&lt;0.476&gt;}</code>
+
+  <p><em>Step 4:</em> Start the client side:</p>
+  <code type="erl">1 client> ssl:start().
+  ok</code>
+
+  <code type="erl">2 client> {ok, Socket} = gen_tcp:connect("localhost", 9999,  [], infinity).</code>
+
+  <p><em>Step 5:</em> Do the TLS handshake:</p>
+  <code type="erl">4 server> {ok, TLSSocket} = ssl:handshake(Socket, [{verify, verify_peer},
+  {cacertfile, "cacerts.pem"},
+  {certs_keys, [#{certfile => "cert.pem", keyfile => "key.pem"}]}]).
+  {ok,{sslsocket,[...]}}</code>
       
-      <p><em>Step 7:</em> Upgrade to a TLS connection. The client and
-      server must agree upon the upgrade. The server must call
-      <c>ssl:handshake/2</c> before the client calls
-      <c>ssl:connect/3.</c></p>
+  <p><em>Step 6:</em> Upgrade to a TLS connection. The client and
+  server must agree upon the upgrade. The server must be prepared to be a TLS server before the client
+  can do a successful connect.</p>
 
-      <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
-{certfile, "cert.pem"}, {keyfile, "key.pem"}], infinity).
+  <code type="erl">3 client>{ok, TLSSocket} = ssl:connect(Socket, [{cacertfile, "cacerts.pem"},
+  {certs_keys, [#{certfile => "cert.pem", keyfile => "key.pem"}]}], infinity).
 {ok,{sslsocket,[...]}}</code>
-      
-      <p><em>Step 8:</em> Send a message over TLS:</p>
+
+      <p><em>Step 7:</em> Send a message over TLS:</p>
       <code type="erl">4 client> ssl:send(TLSSocket, "foo").
-ok</code>
+      ok</code>
       
-      <p><em>Step 9:</em> Set <c>active true</c> on the TLS socket:</p>
-      <code type="erl">4 server> ssl:setopts(TLSSocket, [{active, true}]).
-ok</code>
+      <p><em>Step 8:</em> Set <c>active once</c> on the TLS socket:</p>
+      <code type="erl">5 server> ssl:setopts(TLSSocket, [{active, once}]).
+      ok</code>
       
-      <p><em>Step 10:</em> Flush the shell message queue to see that
+      <p><em>Step 9:</em> Flush the shell message queue to see that
       the message was sent on the client side:</p>
```

Comment:
One more picky comment. I think this sentence 
_Flush the shell message queue to see that
      the message was sent on the client side:_

could be made more precise into:
_Flush the shell queue to see that
      the message was received on the server side:_

Reply:
Reformulated!